src/NewsService.Mappers/ResponsesMappers/NewsResponseMapper.cs

```diff
@@ -0,0 +1,57 @@
+﻿using LT.DigitalOffice.Broker.Requests;
+using LT.DigitalOffice.Broker.Responses;
+using LT.DigitalOffice.Kernel.Broker;
+using LT.DigitalOffice.Kernel.Exceptions;
+using LT.DigitalOffice.NewsService.Mappers.ResponsesMappers.Interfaces;
+using LT.DigitalOffice.NewsService.Models.Db;
+using LT.DigitalOffice.NewsService.Models.Dto.ModelResponse;
+using LT.DigitalOffice.NewsService.Models.Dto.Models;
+using MassTransit;
+using Microsoft.AspNetCore.Mvc;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.NewsService.Mappers.ResponsesMappers
+{
+    public class NewsResponseMapper : INewsResponseMapper
+    {
+        private IRequestClient<IGetUserDataRequest> _client;
+        public NewsResponseMapper(
+            [FromServices] IRequestClient<IGetUserDataRequest> client)
+        {
+            _client = client;
+        }
+        public async Task<NewsResponse> Map(DbNews value)
+        {
+            if (value == null)
+            {
+                throw new BadRequestException();
+            }
+
+            var authorRequest = IGetUserDataRequest.CreateObj(value.AuthorId);
+
+            var authorResponse = await _client.GetResponse<IOperationResult<IGetUserDataResponse>>(authorRequest);
```

Comment:
put GetResponse for author and sender in try catch. do not throw exception in catch, just log error

Reply:
done